core/ir/aarch64/instr_create.h

```diff
@@ -2532,5 +2533,16 @@ enum {
 #define INSTR_CREATE_umull2_vector(dc, Rd, Rm, Rn, width) \
     instr_create_1dst_3src(dc, OP_umull2, Rd, Rm, Rn, width)
 
+/**
+ * Creates an LDR (immediate) instruction.
+ * \param dc      The void * dcontext used to allocate memory for the instr_t.
+ * \param Rd      The output register.
+ * \param mem     The source memory opnd.
+ * \param imm     The integer constant opnd_t operand.
+ */
+#define INSTR_CREATE_ldr_imm(dc, Rd, mem, imm)                                      \
+    instr_create_2dst_3src((dc), OP_ldr, (Rd), opnd_create_reg(opnd_get_base(mem)), \
```

Comment:
See above: I would expect to not have a 2nd destination based on the name, which in the manual just means it has a displacement.  I'm suggesting using `INSTR_CREATE_ldr_wbimm` to match ARM and indicate it's a "writeback" operation which modifies the base reg (and making that clear in the doxygen above).  @AssadHashmi seem reasonable?

Reply:
Yes

Reply:
Will do. Thanks for the feedback.